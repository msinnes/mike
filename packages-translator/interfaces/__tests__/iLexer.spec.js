const { isInterface, loadClass } = require('@mike/class');
const LexerContext = require('@mike/translator-classes/LexerContext');

const iLexer = require('../iLexer');

const TestableLexerContext = loadClass(function() {}).extend(LexerContext);

describe('iLexer', () => {
  it('should be an interface', () => {
    expect(isInterface(iLexer)).toBe(true);
  });

  it('should accept a valid config', () => {
    expect(() => {
      iLexer.ensure({
        tokenizers: [],
        skips: [],
        ctx: new TestableLexerContext(''),
      });
    }).not.toThrow();
  });

  it('should throw an error if the config is invalid', () => {
    expect(() => {
      iLexer.ensure({});
    }).toThrowErrorMatchingSnapshot();
  });
});
