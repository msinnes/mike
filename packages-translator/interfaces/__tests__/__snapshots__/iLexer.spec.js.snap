// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iLexer should throw an error if the config is invalid 1`] = `
"ClassError: Runtime Interface Error
tokenizers: Expected Array
skips: Expected Array
ctx: Expected LexerContext"
`;
